services/management/grpc/actions_server.go

```diff
@@ -294,10 +295,58 @@ func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managemen
 	}, nil
 }
 
-// StartPTMySQLSummaryAction starts pt-mysql-summary action.
+func pointerToAgentType(agentType models.AgentType) *models.AgentType {
+	return &agentType
+}
+
+// StartPTMySQLSummaryAction starts pt-mysql-summary action and returns the pointer to the response message
 //nolint:lll
-func (s *actionsServer) StartPTMySQLSummaryAction(context.Context, *managementpb.StartPTMySQLSummaryActionRequest) (*managementpb.StartPTMySQLSummaryActionResponse, error) {
-	panic("not implemented yet")
+func (s *actionsServer) StartPTMySQLSummaryAction(ctx context.Context, req *managementpb.StartPTMySQLSummaryActionRequest) (*managementpb.StartPTMySQLSummaryActionResponse, error) {
+	// Need to get the service id's pointer to retrieve the list of agent pointers therefrom
+	// to get the particular agentID from the request.
+	service, err := models.FindServiceByID(s.db.Querier, req.ServiceId)
+	if err != nil {
+		return nil, err
+	}
+
+	agents, err := models.FindPMMAgentsRunningOnNode(s.db.Querier, service.NodeID)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "No pmm-agent running on this node")
+	}
+
+	agentID, err := models.FindPmmAgentIDToRunAction(req.PmmAgentId, agents)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := models.CreateActionResult(s.db.Querier, agentID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Exporters to be filtered by service ID and agent type
+	agentFilterStruc := models.AgentFilters{ServiceID: req.ServiceId, AgentType: pointerToAgentType(models.MySQLdExporterType)}
+
+	// Need to get the mysql exporters to get the username and password therefrom
+	pExportersStruc, err := models.FindAgents(s.db.Querier, agentFilterStruc)
```

Comment:
The purpose is to distinguish the pointer to structure from a structure on stack. Go language's compiler will not find an error if you access a member of a structure on stack of or from a pointer to a structure. It is struc.member or pointer.member (which may be dangerous if you find you have a structer and don't initialize the pointer). You'll find it at runtime that the something went from (better case is segmentation fault). Therefore from the safety purposes, it is good that everybody sees it is a pointer.
I've worked with safety critical proprietary software so here I forget that this is just an open-source where safety may not be that important.
Was it just a question from your side or do you want me to change it to that name suggested by yourself?

Reply:
As I said before, we don't use Hungarian notation

Reply:
Yes, you did.
The p stands for pointer, not for the type it points at.

Reply:
So? `pX` is part of Hungarian notation too: https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-6.0/aa260976(v=vs.60)

Reply:
I didn't know p was part of it because I saw it with Linux code where Linux coders are not to much into Windows.
Do you want me to type Ptr at the end?